### PR TITLE
Model object batch padding and gradient functions

### PR DESCRIPTION
**Pull request by joegomes**
This PR adds:
- Moves support for batch padding from `Model` object to `TensorflowGraphModel` object.  Constant batch size is necessary to use Tensorflow's functional programming API (tf.map_fn).  This change moves the `pad_batches` bool from function arguments into self since batch padding should be decided during model initialization rather than at fit/predict time.  This also makes the `fit()` and `predict()` API consistent between sklearn and tf models. 
- Removes gradient error evaluators in `Model` object.  These functions needed to be included in the `Model` object previously, due to the removal of `TensorflowModel` wrapper this is no longer needed.  The functions have been migrated into their respective classes off-repo, to be added back into Deepchem in a later PR.
- Removes fit_transformers from `Model`.  Specific classes for performing on-the-fly transformation will be added in a later PR.

### CODE REVIEW

**Comment by rbharath** [comment]
@joegomes This is an elegant change! I think moving the `pad_batches` argument into the constructor is a good choice. It removes a lot of the messiness of dealing with models that need padding. LGTM.
